### PR TITLE
kssd: 1.1 -> 2.21; enable aarch64-linux support

### PR DESCRIPTION
**Pull request by natsukium**
###### Description of changes

Diff: https://github.com/yhg926/public_kssd/compare/v1.1...v2.21

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by wegank** [comment]
@ofborg build kssd kssd.passthru.tests

**Review by kirillrdy — APPROVED** [review]
Result of `nixpkgs-review pr 243431` run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>kssd</li>
  </ul>
</details>


the segfaults on ofborg, is it possible, just not enough ram ?

edit: i ran my test on machine with a lot of ram

File: pkgs/applications/science/biology/kssd/default.nix [hunk 1]
```diff
@@ -1,38 +1,46 @@
...
-      install -vD kssd $out/bin/kssd
+    runHook preInstall
+
+    install -vD kssd $out/bin/kssd
+
+    runHook postInstall
   '';
 
+  passthru.tests = {
+    simple = runCommand "${finalAttrs.pname}-test" { } ''
```
Comment (kirillrdy) [inline pkgs/applications/science/biology/kssd/default.nix:31]: ok, i can reproduce segfault locally

```
/build/.attr-0l2nkwhif96f51f4amnlf414lhl4rv9vh8iffyp431v6s28gsr90: line 2:     6 Aborted                 (core dumped) /nix/store/a56iln8xbjz38ixfjz2w804y0mpix4kd-kssd-2.21/bin/kssd dist -L /nix/store/97nwzkrxz4ir9rp6b1nrar5mrnp3l0ha-source/shuf_file/L3K10.shuf -r /nix/store/97nwzkrxz4ir9rp6b1nrar5mrnp3l0ha-source/test_fna/seqs1 -o $out/reference
error: builder for '/nix/store/ywfwn7ix85h6xq6v44x83i25mm312x9r-kssd-test.drv' failed with exit code 134;
       last 2 log lines:
       > *** buffer overflow detected ***: terminated
       > /build/.attr-0l2nkwhif96f51f4amnlf414lhl4rv9vh8iffyp431v6s28gsr90: line 2:     6 Aborted                 (core dumped) /nix/store/a56iln8xbjz38ixfjz2w804y0mpix4kd-kssd-2.21/bin/kssd dist -L /nix/store/97nwzkrxz4ir9rp6b1nrar5mrnp3l0ha-source/shuf_file/L3K10.shuf -r /nix/store/97nwzkrxz4ir9rp6b1nrar5mrnp3l0ha-source/test_fna/seqs1 -o $out/reference
       For full logs, run 'nix log /nix/store/ywfwn7ix85h6xq6v44x83i25mm312x9r-kssd-test.drv'.

```

File: pkgs/applications/science/biology/kssd/default.nix [hunk 2]
```diff
@@ -1,38 +1,46 @@
...
+    '';
+  };
+
   meta = with lib; {
     description = "K-mer substring space decomposition";
     license     = licenses.asl20;
     homepage    = "https://github.com/yhg926/public_kssd";
     maintainers = with maintainers; [ unode ];
-    platforms = [ "x86_64-linux" ];
+    platforms = platforms.linux;
```
Comment (kirillrdy) [inline pkgs/applications/science/biology/kssd/default.nix:44]: @natsukium can you please set `meta.mainProgram`

File: pkgs/applications/science/biology/kssd/default.nix (hunk 1 above)
Comment (natsukium) [reply pkgs/applications/science/biology/kssd/default.nix:31]: I have tried changing the parameters and truncating the test data, but I cannot reduce the memory usage below 2 GB.
CI is grayed out, but would such a test be acceptable?

Comment (kirillrdy) [reply pkgs/applications/science/biology/kssd/default.nix:31]: this fixes segfault https://github.com/yhg926/public_kssd/pull/11

File: pkgs/applications/science/biology/kssd/default.nix [hunk 3]
```diff
@@ -1,38 +1,56 @@
...
   };
 
   patches = [
-    # Pull upstream patch for -fno-common toolchain support:
-    #   https://github.com/yhg926/public_kssd/pull/9
     (fetchpatch {
-      name = "fno-common.patch";
-      url = "https://github.com/yhg926/public_kssd/commit/cdd1e8aae256146f5913a3b4c723b638d53bdf27.patch";
-      sha256 = "sha256-HhaTRqPfKR+ouh0PwEH6u22pbuqbX2OypRzw8BXm0W4=";
+      name = "allocate-enough-memory.patch";
```
Comment (kirillrdy) [inline pkgs/applications/science/biology/kssd/default.nix:23]: @natsukium can you add comment referencing PR

**Comment by kirillrdy** [comment]
aarch64-linux has passed, i think we are good to merge

File: pkgs/applications/science/biology/kssd/default.nix (hunk 1 above)
Comment (natsukium) [reply pkgs/applications/science/biology/kssd/default.nix:40]: Awesome! Thank you!

File: pkgs/applications/science/biology/kssd/default.nix (hunk 3 above)
Comment (natsukium) [reply pkgs/applications/science/biology/kssd/default.nix:24]: done

**Review by kirillrdy — APPROVED** [review]

**Pull request merged**